docusaurus/docs/iOS/client/controllers/channels.md

```diff
@@ -55,9 +55,26 @@ let compoundFilter: Filter<ChannelListFilterScope> = .and([
 The `query.sort` is an array of sorting options. Sorting options are applied based on their order in the array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
 
 ### 2. Create a controller
-
+#### With auto Channel Filtering
 The controller can be created using `ChatClient`:
 ```swift
+let controller = ChatClient.shared.channelListController(query: query)
+```
+The behaviour is controlled from the `ChatClientConfig` and more specifically the `isChannelAutomaticFilteringEnabled` flag. 
+
+When the flag is enabled (`true`), then the new/updated channels are being filtered automatically, by matching the `query.filter`.
+By default, the SDK will handle filtering the channels as they get updated and will only return to the controller **only** the channels that match the provider `query.filter`. 
+This means, that the controller always receives channels that match its `query.filter` and then the controller is responsible for deciding if the channels need to be linked/unlinked.
+
+When the flag is disabled (`false`), then the new/updated channels won't be filtered as the arrive, and the controller will be responsible for filtering them first and confirm that they
+match its `query.filter` and then decide if the filtered channels need to be linked/unlinked.
+
+:::warning
+Custom filters are not currently supported. If you are already defining and using your own filters, then the initialization option `With Channel filtering closure` will be a better fit more your needs.
+:::
+
+#### With Channel filtering closure
+```swift
 let controller = ChatClient.shared.channelListController(query: query, filter: { channel in
     // the filtering logic goes here.
     return true
```

Comment:
We definitely don't want this anymore. Customers have a really hard time understanding what is a filter closure since we don't provide a real example.